include/AutomatableModel.h

```diff
@@ -80,6 +82,32 @@ class LMMS_EXPORT AutomatableModel : public Model, public JournallingObject
 
 	virtual ~AutomatableModel();
 
+	// Implement those by using the MODEL_IS_VISITABLE macro
+	virtual void accept(ModelVisitor& v) = 0;
+	virtual void accept(ConstModelVisitor& v) const = 0;
+	// use this to make subclasses visitable
+#define MODEL_IS_VISITABLE \
+	void accept(ModelVisitor& v) override { v.visit(*this); } \
+	void accept(ConstModelVisitor& v) const override { v.visit(*this); }
```

Comment:
Why did you add the macro definition into the class? I think you may pull out the definition like `mapPropertyFromModel`.

Reply:
I wanted to make them look like class members, but now I think it looks better if they are next to `mapPropertyFromModel` .